frontend/src/components/homepage/Filter.tsx

```diff
@@ -32,39 +34,65 @@ const Filter = ({
   setIsTranslator,
   isDisabled = false,
 }: FilterProps) => {
+  const approvedLanguages = role
+    ? approvedLanguagesTranslation
+    : approvedLanguagesReview;
+
   const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
     setLanguage(event.target.value);
   };
+
   const handleRoleChange = (nextRole: string) => {
+    const newApprovedLanguages =
+      nextRole === "Translator"
+        ? approvedLanguagesTranslation
+        : approvedLanguagesReview;
+    const newLanguage = Object.keys(newApprovedLanguages)[0];
```

Comment:
When switching between translator<>reviewer, do we want to do a check first to see if the user has permission to translate/review in the same language that's currently selected instead of defaulting to the first language in the array? At least based on the current seed data, all the languages a user is able to review, they are able to translate as well--I was a bit confused when testing and switching around from reviewer to translator to see the language switch, but also not sure if that'd just overcomplicate with limited user experience value add

Reply:
good point - I think what jenn is describing is more user friendly

Reply:
yup I agree! i'll add that in